Description: Fix certificate hostname validation
 Handle update to geventhttpclient library that enabled validation of hostname
 in server certificate
Origin: OpenContrail, https://raw.githubusercontent.com/Juniper/contrail-third-party/master/ifmap-python-patch2.diff
Forwarded: no
Last-Update: 2014-04-29
Author: Deepinder Setia <dsetia@juniper.net>
Reviewed-by: Sebastien Badia <seb@sebian.fr>

--- ifmap-python-client-0.1.orig/ifmap/client.py
+++ ifmap-python-client-0.1/ifmap/client.py
@@ -92,7 +92,14 @@
 		self.__url = url
                 self.__username = user
                 self.__password = password
-                self._http = HTTPClient(*self.__url, ssl = True,
+                try:
+                    self._http = HTTPClient(*self.__url, ssl = True,
+                                        connection_timeout = None,
+                                        network_timeout = None,
+                                        ssl_options = self.__ssl_options,
+                                        insecure = True)
+                except TypeError:
+                    self._http = HTTPClient(*self.__url, ssl = True,
                                         connection_timeout = None,
                                         network_timeout = None,
                                         ssl_options = self.__ssl_options)
